src/specter.rs

```diff
@@ -140,9 +141,21 @@ impl<T: Transport + Sync + Send> HWI for Specter<T> {
             if new_psbt.inputs[i].tap_key_sig.is_some() {
                 has_signed = true;
                 psbt.inputs[i].tap_key_sig = new_psbt.inputs[i].tap_key_sig;
+            } else {
+                // Specter does not populate PSBT_TAP_KEY_SIG at v1.9.0
+                // see https://github.com/cryptoadvance/specter-diy/issues/277#issuecomment-2183906271
+                if let Some(witness) = &new_psbt.inputs[i].final_script_witness {
+                    let witness = witness.to_vec();
+                    if !witness.is_empty() {
+                        let sig = &witness[0];
```

Comment:
I think you can change these three lines for:
```
if let Some(sig) = witness.nth(0)
```

Reply:
done